fastlane/lanes/localization.rb

```diff
@@ -223,50 +266,80 @@
   #
   desc 'Downloads the localized metadata (for App Store Connect) from GlotPress for the Jetpack app'
   lane :download_jetpack_localized_app_store_metadata do
-    # @FIXME: Replace this whole lane with a call to the future replacement of `gp_downloadmetadata` once it's implemented in the release-toolkit (see paaHJt-31O-p2).
-
-    # No need to `cd` into `fastlane` because of how Fastlane manages its paths internally.
-    sh './download_metadata.swift jetpack'
+    metadata_directory = File.join(PROJECT_ROOT_FOLDER, 'fastlane', 'jetpack_metadata')
 
-    # @TODO: Make the `fastlane/jetpack_metadata/en-US/release_notes.txt` path be the source of truth for the original copies in the future.
+    # FIXME: We should make the `fastlane/jetpack_metadata/default/release_notes.txt` path be the source of truth for the original copies in the future.
     # (will require changes in the `update_appstore_strings` lane, the Release Scenario, the MC tool to generate the announcement post…)
     #
-    # In the meantime, since GlotPress doesn't have the `en-US` notes because those are the ones used as originals, just copy the file to the right place for `deliver` to find
-    metadata_directory = File.join(PROJECT_ROOT_FOLDER, 'fastlane', 'jetpack_metadata')
-    release_notes_source = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'Jetpack', 'Resources', 'release_notes.txt')
-    FileUtils.cp(release_notes_source, File.join(metadata_directory, 'en-US', 'release_notes.txt'))
+    # In the meantime, just copy the file to the right place for `deliver` to find, for the `default` pseudo-locale which is used as fallback
+    release_notes_source = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'Resources', 'release_notes.txt')
+    FileUtils.cp(release_notes_source, File.join(metadata_directory, 'default', 'release_notes.txt'))
+
+    # Download metadata translations from GlotPress
+    download_localized_app_store_metadata(
+      glotpress_project_url: GLOTPRESS_JETPACK_METADATA_PROJECT_URL,
+      locales: JETPACK_METADATA_GLOTPRESS_LOCALE_CODES,
+      metadata_directory: metadata_directory,
+      commit_message: 'Update Jetpack metadata translations'
+    )
+  end
+
+  def download_localized_app_store_metadata(glotpress_project_url:, locales:, metadata_directory:, commit_message:)
+    # FIXME: Replace this with a call to the future replacement of `gp_downloadmetadata` once it's implemented in the release-toolkit (see paaHJt-31O-p2).
 
-    metadata_path = File.join(metadata_directory, '**', '*.txt')
-    git_add(path: metadata_path, shell_escape: true)
+    locales_map = GLOTPRESS_TO_ASC_METADATA_LOCALE_CODES.slice(*locales)    
+    what_new_key = "v#{ios_get_app_version}-whats-new"
+    gp_downloadmetadata(
+      project_url: glotpress_project_url,
+      target_files: APPSTORECONNECT_METADATA_KEYS_TO_FILES_MAP,
+      locales: locales_map,
+      download_path: metadata_directory
+    )
+    files_to_commit = [File.join(metadata_directory, '**', '*')]
```

Comment:
Any particular reason for dropping the `.txt` extension what was in the previous version (at line `-265`)?

Just curious. I think it's fine. Realistically, there shouldn't be other files in those locations.

Reply:
Good catch; it's a leftover from previous attempts at using `*` to include `.gitkeep` files as well, until I realized after testing it that `*` ignores dotfiles so that didn't work.

Now that I've moved to the new approach with `files_to_commit.append(…)` to cover the case of `.gitkeep` files, we could restore the `*.txt` here from before indeed.

Reply:
```suggestion
    files_to_commit = [File.join(metadata_directory, '**', '*.txt')]
```